app/models/mixins/authentication_mixin.rb

```diff
@@ -443,8 +443,13 @@ def assign_nested_endpoint(attributes)
   end
 
   def assign_nested_authentication(attributes)
-    record = authentications.where(:authtype => attributes['authtype']).first_or_initialize
     klass = authentication_class(attributes)
+    record = authentications.find_by(:authtype => attributes['authtype'])
+    if record.nil?
+      record = klass.new(:resource => self)
+      authentications << record
```

Comment:
so, we only want to push newly created records, right?  just checking

Reply:
That's correct, when we are editing an authentication record the `:type` has been saved already so we don't have the same issue as initial auth creation so we're good to go right to `assign_attributes`